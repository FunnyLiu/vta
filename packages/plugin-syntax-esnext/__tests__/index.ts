import path from "path";
import { resolveConfig } from "vta";

/* eslint-disable no-param-reassign */
function formatBabelPluginOrPreset(p) {
  if (typeof p === "string") {
    return path.relative(__dirname, p).replace(/\\/g, "/");
  }
  if (Array.isArray(p)) {
    p[0] = formatBabelPluginOrPreset(p[0]);
    return p;
  }
  return p;
}

function formatBabelConfig(babelConfig) {
  ["presets", "plugins"].forEach((c) => {
    babelConfig[c] = babelConfig[c].map((p) => formatBabelPluginOrPreset(p));
  });
  return babelConfig;
}

describe("plugin-syntax-esnext", () => {
  it("webpack", () => {
    process.env.VTA_JEST_TEST = "true";
    const babelConfig = resolveConfig("babel", path.resolve(__dirname, "./data/projects/webpack"));
    expect(JSON.stringify(formatBabelConfig(babelConfig), null, 2)).toMatchSnapshot();
  });
  it("node", () => {
    process.env.VTA_JEST_TEST = "true";
    const babelConfig = resolveConfig("babel", path.resolve(__dirname, "./data/projects/node"));
    expect(JSON.stringify(formatBabelConfig(babelConfig), null, 2)).toMatchSnapshot();
  });
});
