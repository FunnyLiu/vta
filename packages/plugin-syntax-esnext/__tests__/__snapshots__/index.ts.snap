// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin-syntax-esnext node 1`] = `
"{
  \\"presets\\": [
    [
      \\"../../../node_modules/@babel/preset-env/lib/index.js\\",
      {
        \\"modules\\": \\"commonjs\\",
        \\"useBuiltIns\\": \\"usage\\",
        \\"corejs\\": 3
      }
    ]
  ],
  \\"plugins\\": []
}"
`;

exports[`plugin-syntax-esnext webpack 1`] = `
"{
  \\"presets\\": [
    [
      \\"../../../node_modules/@babel/preset-env/lib/index.js\\",
      {
        \\"modules\\": false,
        \\"useBuiltIns\\": false
      }
    ]
  ],
  \\"plugins\\": [
    [
      \\"../../../node_modules/@babel/plugin-transform-runtime/lib/index.js\\",
      {
        \\"useESModules\\": true,
        \\"corejs\\": 3,
        \\"version\\": \\"7.8.0\\"
      }
    ]
  ]
}"
`;
